src/tribler-gui/tribler_gui/dialogs/confirmationdialog.py

```diff
@@ -10,7 +10,7 @@
 
 
 class ConfirmationDialog(DialogContainer):
-    button_clicked = pyqtSignal(int)
+    button_clicked = pyqtSignal(DialogContainer, int)
```

Comment:
Do we really have to pass this `DialogContainer` thing everywhere?

Reply:
Now that we removed the assignment of dialogs to object attributes, the callback sometimes needs the dialog object to complete its jobs. For example, a callback might require some of the written text in the dialog and check it before the dialog can be closed (this happens with the `AddTagsDialog`). Passing the dialog to the signal callback is an adequate approach to achieve this (but maybe I'm missing a much simpler alternative?).

Reply:
In general, functions/methods/procedures should avoid _extracting_ info from arguments. Otherwise, it is unclear what _exact_ part of the info the function expects from the container argument passed to it. If a child dialog class extends the DialogContainer class, it should overload the signal definition with exact used arguments. E.g.

```python
# Parent class
class DialogContainer:
button_clicked = pyqtSignal(int)

# Child class
InputDialogContainer(DialogContainer):
button_clicked = pyqtSignal(int, str)
```

Reply:
In this situation, the callbacks invoked when clicking one of the buttons might require arbitrary data from the dialog (especially when we create some custom dialogs with additional attributes). For example, a dialog button callback could require access to the text in the dialog input field. Now that we do not bind the dialog to a class attribute anymore, passing the dialog as part of the pyqtSignal invoked when clicking one of the buttons in the dialog is an alternative approach to make the callback aware of the appropriate dialog.

An alternative solution I was thinking of is binding the dialog to a lambda function when connecting the signal to the callback. So for example, instead of writing:

```
connect(dialog.button_clicked, self.on_start_download_action)
```

We would write something like:

```
connect(dialog.button_clicked, lambda action: self.on_start_download_action(dialog, action))
```

However, I believe this is not very readable and I prefer passing the dialog as part of the Qt signal emit. If you have a cleaner solution, please let me know and I can consider rewriting the callbacks.

Reply:
>  the callbacks invoked when clicking one of the buttons might require arbitrary data from the dialog 
> to make the callback aware of the appropriate dialog.

I'd say the solution should be to create a subclass of the Dialog class for each kind of dialog with callbacks being methods of the said subclass. This way we have a very clear understanding of what the callback does in the context of the dialog.